openedx/features/content_type_gating/tests/test_access.py

```diff
@@ -0,0 +1,108 @@
+            )
+            self.welcome = ItemFactory.create(
+                parent=self.chapter,
+                display_name='Welcome'
+            )
+            ItemFactory.create(
+                parent=self.course,
+                category='chapter',
+                display_name='Week 1'
+            )
+            self.chapter_subsection = ItemFactory.create(
+                parent=self.chapter,
+                category='sequential',
+                display_name='Lesson 1'
+            )
+            chapter_vertical = ItemFactory.create(
+                parent=self.chapter_subsection,
+                category='vertical',
+                display_name='Lesson 1 Vertical - Unit 1'
+            )
+            self.problem = ItemFactory.create(
+                parent=chapter_vertical,
+                category='problem',
+                display_name='Problem - Unit 1 Problem 1',
+                graded=True,
+            )
+            self.dragndrop = ItemFactory.create(
+                parent=chapter_vertical,
+                category='drag-and-drop-v2',
+                display_name='Drag Problem - Unit 1 Problem 2',
+                graded=True,
+            )
+            self.ora = ItemFactory.create(
+                parent=chapter_vertical,
+                category='openassessment',
+                display_name='ORA - Unit 1 Problem 3',
+                graded=True,
+            )
+            self.done = ItemFactory.create(
+                parent=chapter_vertical,
+                category='done',
+                display_name='Done - Unit 1 Problem 4',
+                graded=True,
+            )
+            self.edx_sga = ItemFactory.create(
+                parent=chapter_vertical,
+                category='edx_sga',
+                display_name='SGA - Unit 1 Problem 5',
+                graded=True,
+            )
+
+    def setUp(self):
+        super(TestProblemTypeAccess, self).setUp()
+        self.audit_user = UserFactory.create()
+        self.enrollment = CourseEnrollmentFactory.create(user=self.audit_user, course_id=self.course.id, mode='audit')
+
+    @ddt.data(
+        'problem',
+        'dragndrop',
+        'ora',
+        'done',
+        'edx_sga',
+    )
+    @override_settings(FIELD_OVERRIDE_PROVIDERS=(
+        'openedx.features.content_type_gating.field_override.ContentTypeGatingFieldOverride',))
+    @override_waffle_flag(CONTENT_TYPE_GATING_FLAG, True)
+    def test_audit_fails_access_graded_problems(self, problem_type):
+        fake_request = Mock()
+        with patch('courseware.access_response.IncorrectPartitionGroupError.__init__') as mock_access_error:
+            mock_access_error.return_value = IncorrectPartitionGroupError
+            with self.assertRaises(Http404):
+                load_single_xblock(fake_request, self.audit_user.id, unicode(self.course.id),
+                                   unicode(getattr(self, problem_type).scope_ids.usage_id), course=None)
```

Comment:
This is a nit picky thing, but i think its more readable to write multi line calls like this:
```
            load_single_xblock(
                request=fake_request,
                user_id=self.audit_user.id,
                course_id=unicode(self.course.id),
                usage_key_string=unicode(getattr(self, problem_type).scope_ids.usage_id),
                course=None
            )
```